src/renderer/components/+workloads-pods/pod-container-env/env.tsx

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import { withInjectables } from "@ogre-tools/injectable-react";
+import { sortBy } from "lodash";
+import { autorun } from "mobx";
+import React, { useEffect } from "react";
+import type { ConfigMapStore } from "../../+config-maps/store";
+import configMapStoreInjectable from "../../+config-maps/store.injectable";
+import { SecretKey } from "../../+config-secrets/secret-key";
+import type { SecretStore } from "../../+config-secrets/store";
+import secretStoreInjectable from "../../+config-secrets/store.injectable";
+import type { EnvVar } from "../../../../common/k8s-api/endpoints";
+import type { Logger } from "../../../../common/logger";
+import loggerInjectable from "../../../../common/logger.injectable";
+
+export interface ContainerEnvProps {
+  env: EnvVar[];
+  namespace: string;
+}
+
+interface Dependencies {
+  secretStore: SecretStore;
+  configMapStore: ConfigMapStore;
+  logger: Logger;
+}
+
+const NonInjectedContainerEnv = ({
+  env,
+  namespace,
+  configMapStore,
+  secretStore,
+  logger,
+}: ContainerEnvProps & Dependencies) => {
+  useEffect(() => autorun(() => {
+    for (const { valueFrom } of env) {
+      if (valueFrom?.configMapKeyRef) {
+        const { configMapKeyRef } = valueFrom;
+
+        configMapStore
+          .load({ name: configMapKeyRef.name, namespace })
+          .catch(error => logger.warn(`[CONTAINER-ENV]: failed to load ConfigMap ${configMapKeyRef.name} in ns=${namespace}`, error));
+      }
+
+      if (valueFrom?.secretKeyRef) {
+        const { secretKeyRef } = valueFrom;
+
+        secretStore
+          .load({ name: secretKeyRef.name, namespace })
+          .catch(error => logger.warn(`[CONTAINER-ENV]: failed to load Secret ${secretKeyRef.name} in ns=${namespace}`, error));
+      }
+    }
+  }), []);
```

Comment:
Render-time side-effect. I know you just moved it here, so I don't see this as something you should fix now. 

Real solution would be that this component gets the values that it needs as dependency and that dependency would do the loading etc. Case where `asyncComputed` can be utilized to completely remove the `useEffect` and more importantly `autorun`.

But as said, not now, since you are just extracting it as own component.